Public/Src/Engine/Scheduler/FileMonitoringViolationAnalyzer.cs

```diff
@@ -873,6 +873,13 @@ private static bool IsAllowedSameContentDoubleWrite(Process secondWriter, FileMa
                             return;
                         }
 
+                        // WriteOnAbsentPathProbe message literaly says "declare an explicit dependency between these pips",
+                        // so don't complain if a dependency already exists (i.e., 'pip' must run after 'related').
+                        if (m_graph.IsReachableFrom(from: related, to: pip))

```

Comment:
> if (m_graph.IsR [](start = 23, length = 16)

Wondering if this actually works reliably for the case of exlusive opaques. Exclusive opaques are scrubbed, but right before the writer runs, so the pip that probes may find the write there

Reply:
In a way, exclusive opaques may suffer from the same problems I was describing on email if the shared opaque scrubber becomes smarter

---
In reply to: [305936799](https://github.com/microsoft/BuildXL/pull/638#discussion_r305936799) [](ancestors = 305936799)